prow/plugins/plugins.go

```diff
@@ -160,6 +168,15 @@ func RegisterGenericCommentHandler(name string, fn GenericCommentHandler, help H
 	genericCommentHandlers[name] = fn
 }
 
+// GenericCommentHandler defines the function contract for a github.GenericCommentEvent handler.
+type ExtendedGenericCommentHandler func(Agent, github.GenericCommentEvent) Status
+
+// RegisterExtendedGenericCommentHandler registers a plugin's github.GenericCommentEvent handler.
+func RegisterExtendedGenericCommentHandler(name string, fn ExtendedGenericCommentHandler, help HelpProvider) {
```

Comment:
Maybe we would want to (somehow) prevent plugins from registering both a normal and extended handler?

Reply:
Prevented by doing the check if the latter is added to extended/not-extended map. If yes, the error is logged and the handler is not added.

Any additional move would require an interface check. Please let me know if that's enough.

Reply:
Do we plan to eventually move all plugins to be an `ExtendedGenericCommentHandler`? Maybe we can rename the existing registerer to `Legacy` to clarify that it shouldn't be used for new plugins?

Reply:
I don't think so. It was introduced to add additional metrics for heavy plugins like `lgtm` and `approve`. I do not see other clear candidates to have the interface extended to have this information. I don't know though, it's my opinion.